src/object.d

```diff
@@ -1431,6 +1431,132 @@ class TypeInfo_Struct : TypeInfo
     assert(!typeid(S).equals(&s, &s));
 }
 
+class RTTypeid(T) : TypeInfo
+if (is(T == struct))
+{
+    override string toString() const { return T.stringof; }
+
+    override bool opEquals(Object rhs)
+    {
+        if (this is rhs)
+            return true;
+        auto s = cast(const RTTypeid!T) rhs;
+        return s !is null;
+    }
+
+    override size_t getHash(scope const void* p) @trusted pure nothrow const
```

Comment:
TypeInfo doesn't require `pure`. This might fail below because toHash can be impure (allthough it obviously should be)

```suggestion
    override size_t getHash(scope const void* p) @trusted nothrow const
```